### PR TITLE
Validate workflowId for call cache diff endpoint

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by geoffjentry** [comment]
:+1:

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala [hunk 1]
```diff
@@ -2,19 +2,21 @@ package cromwell.engine.workflow.lifecycle.execution.callcaching
...
-  case class CallCacheDiffQueryCall(workflowId: String, callFqn: String, jobIndex: Option[Int])
+  case class CallCacheDiffQueryCall(workflowId: WorkflowId, callFqn: String, jobIndex: Option[Int])
 
   private def missingWorkflowError(attribute: String) = s"missing $attribute query parameter".invalidNel
 
   def fromParameters(parameters: Seq[(String, String)]): Validated[NonEmptyList[String], CallCacheDiffQueryParameter] = {
     def extractIndex(parameter: String): Validated[NonEmptyList[String], Option[Int]] = {
       parameters.find(_._1 == parameter) match {
-        case Some((_, value)) => Try(value.toInt) match {
+        case Some((_, value)) => Try(value.trim.toInt) match {
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: Not your fault (NYF): I think we should have a `case None` here that returns an `Invalid`, right now looks like it'd just throw a match exception

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala [hunk 2]
```diff
@@ -29,11 +31,19 @@ object CallCacheDiffQueryParameter {
       }
     }
     
-    val workflowAValidation = extractAttribute("workflowA")
-    val workflowBValidation = extractAttribute("workflowB")
+    def validateWorkflowId(parameter: String) = for {
+      workflowIdString <- extractAttribute(parameter)
+      workflowId <- Try(WorkflowId.fromString(workflowIdString.trim)) match {
+        case Success(id) => id.validNel
+        case Failure(f) => f.getMessage.invalidNel
```
Comment (danbills) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: there's `fromTry` `in cats.data.Validated._` but it needs a `.toValidatedNel`

Comment (danbills) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: there's `fromTry` `in cats.data.Validated._` but it needs a `.toValidatedNel`

Comment (Horneth) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: It says `Invalid UUID string: blabalbla`

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala (hunk 1 above)
Comment (Horneth) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: There is already `case None` below no ? Line 23 https://github.com/broadinstitute/cromwell/pull/2552/files#diff-418822f319fcfc8f92177448f49093f4R23

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala (hunk 2 above)
Comment (Horneth) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: @danbills Even with `fromTry(Try(WorkflowId.fromString(workflowIdString.trim))).toValidatedNel` I get a `Validated[NonEmptyList[Throwable]], WorkflowId]` and I want a `Validated[NonEmptyList[String]], WorkflowId]`, do you know how I can do that ?

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala (hunk 1 above)
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala]: Oh, ok, github fail (it was hiding).

... carry on as you were!

**Pull request merged**